### PR TITLE
feature: build complete updater

### PR DESCRIPTION
**Pull request by akash2237778**
Build complete updater: This container runs a script at specified intervals which removes the executed kaniko pods (in completed state).

### CODE REVIEW

File: build-complete-updater/updater.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import os
+import time
+import requests
+import json
+
+NAMESPACE = "broker"
+BROKER_URL = "http://broker-service.broker"
+BROKER_PORT = "8080"
+SLEEP_INTERVAL = 5
+BUILD_ID_LENGTH = 6
```
Comment (lvarin) [inline build-complete-updater/updater.py:11]: Just to future proof this a bit, this parameter should be able to be configurable with environment variables.

File: build-complete-updater/updater.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+from kubernetes import client, config
+import os
+import time
+import requests
+import json
+
+NAMESPACE = "broker"
+BROKER_URL = "http://broker-service.broker"
+BROKER_PORT = "8080"
+SLEEP_INTERVAL = 5
```
Comment (lvarin) [inline build-complete-updater/updater.py:10]: Just to future proof this a bit, this parameter should be able to be configurable with environment variables.

File: build-complete-updater/updater.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+while True:
+    v1 = client.CoreV1Api()
+    pods = v1.list_namespaced_pod(namespace=NAMESPACE, watch=False)
+    for pod in pods.items:
+        if pod.spec.containers[0].env is not None:
+            if pod.status.phase == "Succeeded":
+                if (
+                    "BUILDNAME" == pod.spec.containers[0].env[0].name
+                    and "ACCESSTOKEN" == pod.spec.containers[0].env[1].name
```
Comment (lvarin) [inline build-complete-updater/updater.py]: This assumes the order in which the environment variables will be in the Pod. There must be a better way to do this. Even this O(n) method is a better approach:

```
def getEnv(env, name):
    for var in env:
        if var.name == name:
            return var.value
```

**Review by lvarin — CHANGES_REQUESTED** [review]

**Review by uniqueg — APPROVED** [review]
If you address @lvarin's comments, it's fine with me.

**Review by lvarin — APPROVED** [review]
LGTM

**Pull request merged**